src/core/ipfs/libp2p.js

```diff
@@ -64,22 +62,7 @@ module.exports = function libp2p (self) {
           maddr = multiaddr(maddr)
         }
 
-        if (!mafmt.IPFS.matches(maddr.toString())) {
-          return callback(new Error('multiaddr not valid'))
-        }
-
-        let ipfsIdB58String
-        maddr.stringTuples().forEach((tuple) => {
-          if (tuple[0] === 421) {
-            ipfsIdB58String = tuple[1]
-          }
-        })
-
-        const id = peerId.createFromB58String(ipfsIdB58String)
-
-        self._libp2pNode.dialByMultiaddr(maddr, (err) => {
-          callback(err, id)
-        })
```

Comment:
Because this was a check to make sure the multiaddr contained the ipfs multihash, but to be fully interop with go, it has to support discovering that post conn is established. Therefore https://github.com/ipfs/js-libp2p-ipfs/pull/16